apis/exoscale/v1/postgresql_types.go

```diff
@@ -0,0 +1,114 @@
+package v1
+
+import (
+	"reflect"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// PostgreSQLParameters are the configurable fields of a PostgreSQL.
+type PostgreSQLParameters struct {
+	Maintenance MaintenanceSpec `json:"maintenance,omitempty"`
+	Backup      BackupSpec      `json:"backup,omitempty"`
+
+	DBaaSParameters `json:",inline"`
+
+	// PGSettings contains additional PostgreSQL settings.
+	PGSettings runtime.RawExtension `json:"pgSettings,omitempty"`
+}
+
+// SizeSpec contains settings to control the sizing of a service.
+type SizeSpec struct {
+	Plan string `json:"plan,omitempty"`
+}
+
+// IPFilter is a list of allowed IPv4 CIDR ranges that can access the service.
+// If no IP Filter is set, you may not be able to reach the service.
+// A value of `0.0.0.0/0` will open the service to all addresses on the public internet.
+type IPFilter []string
+
+// PostgreSQLSpec defines the desired state of a PostgreSQL.
+type PostgreSQLSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+	ForProvider       PostgreSQLParameters `json:"forProvider"`
+}
+
+// PostgreSQLObservation are the observable fields of a PostgreSQL.
+type PostgreSQLObservation struct {
+	NoteStates []NodeState `json:"noteStates,omitempty"`
+	// Version is the version number as returned by Exoscale.
+	Version     string          `json:"version,omitempty"`
+	IPFilter    IPFilter        `json:"ipFilter,omitempty"`
+	Maintenance MaintenanceSpec `json:"maintenance,omitempty"`
+	Backup      BackupSpec      `json:"backup,omitempty"`
+}
+
+// PostgreSQLStatus represents the observed state of a PostgreSQL.
+type PostgreSQLStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+	AtProvider          PostgreSQLObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="State",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].reason"
```

Comment:
This line should probably be deleted.

Reply:
I don't see why?

Reply:
Isn't type `Ready` already configured with the next line?

Reply:
The column "Ready" displays the `.status` field of the Ready condtion, not the `.reason`. The value is given when the Ready is actually `False` and "State" would then display the VM state, e.g. `Rebuilding`, `PoweredOff` etc.

Reply:
Good, thanks for insights.